hieroplatform/src/main/java/org/hiero/sketch/spreadsheet/BucketsDescription1D.java

```diff
@@ -0,0 +1,111 @@
+package org.hiero.sketch.spreadsheet;
+
+import org.checkerframework.checker.nullness.qual.NonNull;
+import java.util.Arrays;
+
+/**
+ * MetaData for one dimensional buckets held by a histogram
+ */
+public class BucketsDescription1D implements IBucketsDescription1D {
+    private final double minValue;
+    private final double maxValue;
+    private final int numOfBuckets;
+    private final double[] boundaries;
+
+    /**
+     * The assumption is that the all buckets are only left inclusive except the right one which
+     * is right inclusive. Boundaries has to be strongly sorted.
+     */
+    public BucketsDescription1D(@NonNull final double[] boundaries) {
+        if (boundaries.length == 0)
+            throw new IllegalArgumentException("Boundaries of buckets can't be empty");
+        if (!isSorted(boundaries))
+            throw new IllegalArgumentException("Boundaries of buckets have to be sorted");
+        this.boundaries = new double[boundaries.length];
+        System.arraycopy(boundaries, 0, this.boundaries, 0, boundaries.length);
+        this.numOfBuckets = boundaries.length - 1;
+        this.minValue = this.boundaries[0];
+        this.maxValue = this.boundaries[this.numOfBuckets];
+    }
+
+    /**
+     * Checks that an array is strongly sorted
+     */
+    private static boolean isSorted(final double[] a) {
+        for (int i = 0; i < (a.length - 1); i++) {
+            if (a[i] > a[i + 1]) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public int indexOf(final double item) {
+        if (item < this.minValue || item > this.maxValue)
+            return -1;
+        if (item == this.maxValue)
+            return this.numOfBuckets - 1;
+        int lo = 0;
+        int hi = this.boundaries.length - 1;
+        while (lo <= hi) {
+            int mid = lo + (hi - lo) / 2;
+            if (item < this.boundaries[mid]) hi = mid ;
+            else if (item >= this.boundaries[mid + 1]) lo = mid;
+            else return mid;
+        }
+        return -1;
+    }
+
+    @Override
+    public double getLeftBoundary(int index) {
+        if (index < 0 || index >= this.numOfBuckets)
+            throw new IllegalArgumentException("Bucket index out of range");
+        return this.boundaries[index];
+    }
+
+    @Override
+    public double getRightBoundary(final int index) {
+        if (index == (this.numOfBuckets - 1))
+            return this.maxValue;
+        return this.getLeftBoundary(index + 1);
+    }
+
+    @Override
+    public int getNumOfBuckets() {
+        return this.numOfBuckets;
+    }
+
+    @Override
+    public double[] getBoundaries() {
```

Comment:
I made a copy so that it is safe to pass on, if a comment is sufficient then I'll change it. It is not needed in any of the classes I wrote. I thought this would be useful for the front end when visualizing the histogram (instead of getting it bucket by bucket) . If not it could be deleted.